src/main/java/com/dynatrace/loadrunner/converter/util/FileReaderUtil.java

```diff
@@ -0,0 +1,125 @@
+package com.dynatrace.loadrunner.converter.util;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.Set;
+
+import org.apache.commons.io.FilenameUtils;
+
+import com.dynatrace.loadrunner.config.InputFiles;
+import com.dynatrace.loadrunner.config.Technology;
+import com.google.common.collect.Sets;
+
+public class FileReaderUtil {
+
+	private static final String USR_FILE_EXT = "usr";
+	private static final String C_BODY_FILE_EXT = "c";
+	private static final String C_GLOBALS_FILE = "globals.h";
+	private static final String JS_BODY_FILE_EXT = "js";
+	private static final String JS_GLOBALS_FILE = "globals.js";
+
+	private static final String REGEX = "[.][^.]+$";
+
+	private FileReaderUtil() {
+		// prevent creation
+	}
+
+	private static String getFileExtension(String fileName) {
+		return FilenameUtils.getExtension(fileName);
+	}
+
+	public static InputFiles findInputFiles(final File folder, Technology technology) {
+		Set<File> allFiles = Sets.newHashSet();
+		findFilesRecursive(folder, allFiles);
+		Collection<File> headerFiles = filterHeaderFiles(allFiles, technology);
+		Collection<File> bodyFiles = filterBodyFiles(allFiles, technology);
+
+		return new InputFiles(headerFiles, bodyFiles, getScriptNameFromPath(allFiles));
+	}
+
+	private static String getScriptNameFromPath(final Collection<File> allFiles) {
+		for (File file : allFiles) {
+			if (USR_FILE_EXT.equals(getFileExtension(file.getName()))) {
+				return file.getName().replaceFirst(REGEX, "");
+			}
+		}
+		return null;
+	}
+
+	private static void findFilesRecursive(final File file, Set<File> output) {
+		if (!file.exists()) {
+			return;
+		}
+		if (file.isDirectory()) {
+			for (final File subFile : Objects.requireNonNull(file.listFiles())) {
+				findFilesRecursive(subFile, output);
+			}
+		} else {
+			output.add(file);
+		}
+	}
+
+	public static Collection<File> getHeaderFiles(String[] input, Technology technology) {
+		return filterHeaderFiles(getFiles(input), technology);
+	}
+
+	public static Collection<File> getBodyFiles(String[] input, Technology technology) {
+		return filterBodyFiles(getFiles(input), technology);
+	}
+
+	private static Collection<File> getFiles(String[] input) {
+		Set<File> output = Sets.newHashSet();
+		for (String inputFile : input) {
```

Comment:
There could be a possible NPE, because it is possible to return null in `UserConfig::split` - please correct me if I'm wrong ;)